### PR TITLE
Add spread element check when the last element is a spread. 

### PR DESCRIPTION
**Pull request by butterunderflow**
The current behavior is: if the last list element expression is a [spread](https://rescript-lang.org/docs/manual/latest/array-and-list#immutable-prepend), the parser will not check the rest element whether contains a spread expression. 
This PR adds a spread element check when the last list element is a spread.
Please see #670 for more details.

### CODE REVIEW

File: tests/parsing/errors/other/expected/spread.res.txt [hunk 1]
```diff
@@ -49,6 +49,20 @@ Explanation: you can't collect a subset of a record's field into its own record,
...
+  tests/parsing/errors/other/spread.res:8:1-3
+
+   6 │ 
+   7 │ let myList = list{...x, ...y}
+   8 │ let list{...x, ...y} = myList
+   9 │ 
+  10 │ type t = {...a}
+
+  Lists can only have one `...` spread, and at the end.
+Explanation: lists are singly-linked list, where a node contains a value and points to the next node. `list[a, ...bc]` efficiently creates a new item and links `bc` as its next nodes. `[...bc, a]` would be expensive, as it'd need to traverse `bc` and prepend each item to `a` one by one. We therefore disallow such syntax sugar.
```
Comment (IwanKaramazow) [inline tests/parsing/errors/other/expected/spread.res.txt:62]: There's a couple of syntax errors here:
* `list[a, ...bc]` doesn't exist
* `[...bc, a]` also doesn't exist

File: src/res_core.ml [hunk 2]
```diff
@@ -3716,6 +3716,14 @@ and parseSpreadExprRegion p =
...
 and parseListExpr ~startPos p =
+  let check_all_non_spread_exp exprs =
+    exprs
+    |> List.map (fun (spread, expr) ->
+           if spread then
+             Parser.err p (Diagnostics.message ErrorMessages.listExprSpread);
+           expr)
+    |> List.rev
+  in
   let listExprs =
```
Comment (cristianoc) [inline src/res_core.ml:3727]: This should be called `listExprsRev` or something.

File: src/res_core.ml [hunk 3]
```diff
@@ -3724,17 +3732,10 @@ and parseListExpr ~startPos p =
   let loc = mkLoc startPos p.prevEndPos in
   match listExprs with
   | (true, expr) :: exprs ->
```
Comment (cristianoc) [inline src/res_core.ml:3734]: Perhaps add comment `true (* spread *)`

**Review by cristianoc — COMMENTED** [review]
Looking good.
A few comments on pre-existing typos in the error message, and suggestions for code cleanup.

Would you add a line to CHANGELOG.md.

**Comment by butterunderflow** [comment]
OK, I'll try that.

**Comment by butterunderflow** [comment]
I think I have done it, please check again. Thank you for reviewing!

**Review by cristianoc — APPROVED** [review]

**Pull request merged**